code/modules/mob/living/say.dm

```diff
@@ -99,6 +99,9 @@ var/list/department_radio_keys = list(
 
 	message = process_chat_markup(message, list("~", "_"))
 
+	if(check_slur(message, usr, "Say"))
```

Comment:
use src isntead of usr here, and also make sure that it interacts correctly with forcesay, so the person being forced to say something isn't marked as the culprit